doc/guides/west/build-flash-debug.rst

```diff
@@ -35,17 +35,25 @@ use :ref:`west config <west-config-cmd>` to configure its behavior.
 This command attempts to "do what you mean" when run from a Zephyr application
 source or build directory:
 
-- When you run ``west build`` in an existing build directory, the board, source
+- When you run ``west build`` and a Zephyr build directory named as the default
```

Comment:
> Since this is the introduction to the command, I think it's important to have clear wording that says what it does "in a nutshell" and "out of the box" here. This change here is hard to understand. I think we can balance this with the need to make it clear there are other options a bit better.
> 
> I've pushed a patch with some ideas, along with applying the other suggestions in these comments, etc:
> 
> https://github.com/mbolivar/zephyr/tree/build-folder-config

I have taken your changes as-is except for a minor fix (debug copy-paste mistake). I think they are better than what I proposed. That said, I still think the default should be `source_dir` instead of `app`, but I see no consensus here and so I'd rather go with what you think makes sense in this particular case.

Reply:
OK, thanks for humoring me and for catching that flash/debug copy/paste error :).